### ISSUE TITLE
enhancement: example messages

### ISSUE THREAD

**Issue by faxm0dem** [balabit/syslog-ng#356]
@bazsi 's ml post on the config debugger gave me an idea: wouldn't it be great to have a similar "example message" mechanism as in patterndb?
The idea is to embed test messages into the config, with information such as through which log paths they ought to flow through. Then a `syslog-ng-tool test` would process all messages and verify they're flowing through the parsers and outputs they're supposed to.

**Comment by ihrwein** [comment]
We have a GSoC project, which has a similar role: https://github.com/balabit/syslog-ng/wiki/GSoC2015-Idea-&-Project-list#helping-troubleshooting

It's not about embedding test messages, but to show the message flow.

**Comment by faxm0dem** [comment]
interesting!

**Comment by gregory094** [comment]
@hkrist has written something like this as the task0 of the GSoC.
You can check it [here](https://github.com/hkrist/syslog-ng/tree/GSoC-task0).

I think your idea is a great feature proposal. What about moving to the the 'feature proposals' column to keep it in the fore?

**Comment by furiel** [comment]
We discussed this old issue, and we decided not to support such functionality in the configuration language. Such feature to be usable, it would need a variety of functionality, that would be difficult to manage inside the configuration language, and also difficult to maintain inside the code.

The resolution is that syslog-ng should provide an e2e test framework, that could be used to test any configuration. This is the purpose of the light framework in syslog-ng, when users can write e2e tests in python: for example https://github.com/balabit/syslog-ng/blob/7c6df27a6a5c7109f21736b6ca0e902beb854381/tests/python_functional/functional_tests/logpath/test_flags_final.py

For now it has a limited feature set, but is improving rapidly.

Closing this ticket in favor of light.
